internal/build/resolver/url.go

```diff
@@ -1,14 +1,15 @@
-package build
+package resolver
 
 import (
+	"context"
 	"fmt"
 	"regexp"
 
 	"github.com/buildkite/cli/v3/internal/build"
 )
 
-func ResolveFromURL(args []string) build.BuildResolverFn {
-	return func() (*build.Build, error) {
+func ResolveFromURL(args []string) BuildResolverFn {
+	return func(context.Context) (*build.Build, error) {
```

Comment:
Eg, if this resolver needed to know the pipeline to be able to resolve the build (ie thinking "most recent build on a branch for a pipeline"), we would bind in a PipelineResolverFn here for the function to use. The signature of the build resolver would remain the same and we don't need to expose that dependency to other resolvers that may not need it (like this one)

Reply:
The API request that is made to determine the "most recent build" would also make use of the context passed in so that it could cancel the request
p